### ISSUE TITLE
Compiling Wyrmsus in windows.

### ISSUE THREAD

**Issue by DinkyDyeAussie** [Andrettin/Wyrmgus#103]
Using CMake to generate the project files for Visual Studio 2015. I have everything directory wise working, except CMake mentions oaml is empty, something else about github and a few other things. I am using older dependancies if that makes a difference.

I really wanna see this game as the recent ahots of it look really good :)

Can anyone help?

**Comment by Andrettin** [comment]
Hey Dinky :)

Maybe you lack the (new) OAML dependency? Here's an issue in which someone had a similar problem:
https://github.com/Andrettin/Wyrmgus/issues/97

**Comment by DinkyDyeAussie** [comment]
So I was able to download the latest trunk and the the latest executable for windows but it won't run.

stderr.txt says this:

`Unsupported tag: ShipNames
stack traceback:
	[C]: in function 'DefineCivilization'
	[string "scripts/civilizations.lua"]:72: in main chunk
	[C]: in function 'Load'
	[string "scripts/stratagus.lua"]:1889: in main chunk
`
and stdout.txt says this:

`..\src\stratagus\script_player.cpp:887: CclDefineCivilization: Unsupported tag: ShipNames`

I don't know what to do. So if you can help me out I'd appreciate it alot :) Looking forward to playing this. It's been so long.

**Comment by Andrettin** [comment]
Hello Dinky, are you sure you have the latest trunk? It sounds like you have an older one before the support for the "ShipNames" field was introduced.

**Comment by DinkyDyeAussie** [comment]
I just downloaded the latest Wyrmsun.exe from your repository and also downloaded the latest trunk and put them in the same folder and ran it. I haven't got the OAML file yet to compile it on my system. I thought I could just download your binary and run it with the latest trunk...

**Comment by DinkyDyeAussie** [comment]
OK so I found the steam version of your game and downloaded it and have been playing the Human Campaign up to the Roman mission. So far I love it! Thing is though, units don't auto attack/defend when being attack by the enemy. That was the only thing/bug I have found so far. Also you put in a day/night cycle? That is so cool! A bit hard to see things at night but still it is very cool :+1: Plus dynamic music? That is also very cool!

Also I discovered that you have sorted out the menu not having the click sound. Wargus doesn't, but the original warcraft 2 does, and wyrmsun does too. I would like to know how you got that working. So many things in Wyrmsun in terms of features that I want in/back in Wargus. Wargus 2.1 behaved exactly like the original warcraft 2, but then they changed everything...

**Comment by Andrettin** [comment]
Oh, the latest release binary only runs with the latest release data, not with the trunk data.

I'm happy you're liking it :) It is strange that the units aren't auto-attacking, I will look into it. In which circumstances did that happen?

About the buttons having a click sound, it was relatively simple to do. For example, look at the present code for image buttons:

``
	function menu:addImageButton(caption, hotkey, x, y, callback)
		local b = ImageButton(caption)
		if (hotkey ~= "") then
			b:setHotKey(hotkey)
		end
		b:setActionCallback(
			function()
				PlaySound("click")
				callback()
			end
		)
		self:add(b, x, y)
		b:setBorderSize(0) -- Andrettin: make buttons not have the borders they previously had
		return b
	end
``

**Comment by DinkyDyeAussie** [comment]
The units not having auto-attack was just something that occured during regular mussion play. Not a single mission had that feature working...at least not for the player.

Whenever I went for the A.I. it auto attacked me. But none of my units were like "oh no, I'm being attacked! Chaaaaarge!" when they were being attacked and minding their own business to begin with.

Hope that helped.

Also I'll look into the lua files and source code when I can and sort out the button click code... 

Also while I'm on this topic, how do I get the button text to highlight when you mouse over them, like what happens in wargus when you hover over the "Menu (F10)" button. So far that's the only button that does it in Wargus...I want them all to do it like original Warcraft 2 did.

I know it's OCD but its the little things like this that really make a game look and feel polished.

**Comment by Andrettin** [comment]
Hmm weird, auto-attack is working correctly in my tests. Maybe the units had other orders already?

About the button text highlight, didn't you make a patch for that based on Wyrmgus code a long time ago (which wasn't implemented into Stratagus because then only cybermind had access and he wasn't active anymore)? I recall something like that.

**Comment by DinkyDyeAussie** [comment]
Yeah something like that but I need to redo it again. I'll look through your code and do it again man. No worries :)

**Comment by DinkyDyeAussie** [comment]
OK so I have changed the code in the guichan.lua to reflect your code for the addButton and now I get the click sound.

Now what I am trying to do is get the text to highlight when you hover...So, how did you do this? What files need modification - both lua and source code files.

Also Since I changed 

`(GameMenuButtonClicked ? MI_FLAGS_CLICKED : 0)`

to

`(UI.MenuButton.Clicked ? MI_FLAGS_CLICKED : 0),`

the "Menu(F10)" button no longer depresses. It is just stuck on the unclicked frame. I have defined all the same parameters you did in CUIButton - both `bool Clicked` and `bool HotkeyPressed`.

Didn't do any other source code modifications though, so I could have missed something critical...Could you please help mate?

**Comment by Andrettin** [comment]
I don't think either of those things are linked to the button text change when hovering.

The best way to go about this would be, IMO, to find that patch you had made and apply it to Stratagus, as it has all the required changes.

Otherwise, you could search the source code for "is_normal", which is the boolean which deals with the text color change for button text in the functions.

**Comment by DinkyDyeAussie** [comment]
I changed the code I mentioned above and that fixed my "pressing" issue. Still searching the code/forums/scripts to find the information I need to get this working. I cannot compile your code in CMAKE as I cannot compile OAML in VS2015 because it "cannot open 'vorbis/codec.h' at all"

Do you have the libraries needed to CMAKE your engines code?

**Comment by DinkyDyeAussie** [comment]
Did a search through your code for "is_normal" did you really add that many to your engine? That's a lot of work.

**Comment by DinkyDyeAussie** [comment]
Besides I can't even remember how I figured out how to get it working last time without editing the engine code...I think I gave up back then as I couldn't upload changes. But now I can and I really want to fix it!

**Comment by DinkyDyeAussie** [comment]
OK so I did it, it's done now. Every button has the text that changes color when you hover over it, I just did what you said about putting "is_normal" in the right spots. I literally copied your code to the letter in this regard and it compiled and ran perfectly.

Cannot thank you enough. I will update the trunk code soon with this so everyone still playing this can enjoy these small but meaningful additions.

**Comment by Andrettin** [comment]
I'm glad it's working fine now :)

**Comment by DinkyDyeAussie** [comment]
Thanks to you mate. :) I need to keep on top on this stuff.

**Comment by DinkyDyeAussie** [comment]
I added all those "is_normal" things to the trunk but it's failing on the build...Could you please check it out?

**Comment by Andrettin** [comment]
In a few lines in /guichan/graphics.cpp (143, 146 and 149), you've put an extra "bool" in the function's code itself. So instead of writing it like this:

```
mFont->drawString(this, text, x, y, bool is_normal);
```

You need to write it like this:

```
mFont->drawString(this, text, x, y, is_normal);
```

**Comment by DinkyDyeAussie** [comment]
Farout...lol. thanks mate.
